src/Integration/NewConnectedMode/ObsoleteConfigurationProvider.cs

```diff
@@ -40,8 +40,8 @@ internal interface IObsoleteConfigurationProviderService : IConfigurationProvide
     {
     }
 
-    [Export(typeof(IConfigurationProvider))]
-    [PartCreationPolicy(CreationPolicy.Any)]
+    [Export(typeof(IObsoleteConfigurationProviderService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
```

Comment:
a h, I thought that was on purpose, btw is that also something that could be tested?

Reply:
It doesn't actually make much difference in this case since the class isn't stateful.

Yes, it could be tested in a couple of ways:
* for singletons, we could have two classes that both import X and check they get the same instance
* we could use reflection to look at the data in the `PartCreationPolicy` attribute and check it has the expected value